cni/network/network.go

```diff
@@ -301,6 +295,35 @@ func addNatIPV6SubnetInfo(nwCfg *cni.NetworkConfig,
 	}
 }
 
+func acquireLockForStore(config *common.PluginConfig, plugin *netPlugin) error {
+	var err error
+	if err = plugin.Store.Lock(true); err != nil {
+		log.Printf("[CNI] Failed to lock store: %v. check if process running", err)
+		if isSafe, _ := plugin.IsSafeToRemoveLock(azureCniName); isSafe {
+			log.Printf("[CNI] Removing lock file as process holding lock exited")
+			if errUninit := releaseLockForStore(plugin); errUninit != nil {
+				log.Errorf("Failed to release lock file, err:%v.\n", errUninit)
+			} else {
+				err = nil
+			}
+		}
+	}
+
+	return err
+}
+
+func releaseLockForStore(plugin *netPlugin) error {
+	if plugin.Store != nil {
+		err := plugin.Store.Unlock(false)
+		if err != nil {
+			log.Printf("[cni] Failed to unlock store: %v.", err)
+			return err
```

Comment:
it doesn't look like you're ever checking the returned error from this function - you're only calling this in defer - so maybe it should not return an error and the log is sufficient.

Reply:
im checking it in another place (inside acquireLockForStore)